internal/connection/connection_pool.go

```diff
@@ -62,11 +62,21 @@ func (cp *ConnectionPool) Delete(key string) {
 	delete(cp.pool, key)
 }
 
-// Get returns map of connections and unlock function to be called
-// after parsing the connections.
-func (cp *ConnectionPool) Get() (*map[string]*Connection, func()) {
-	rlock := cp.rwlock.RLocker()
-	rlock.Lock()
+// Get returns map of connections, filtered with given driverName and optional nodeID.
+func (cp *ConnectionPool) Get(driverName, nodeID string) map[string]*Connection {
```

Comment:
if the connection is not found return error?

Reply:
> if the connection is not found return error?

I think it's better not to return an error here, since connectionPool is just doing basic Get,Put,Delete operation with none of it returning an error. 

controller may need to still return a `not found` error if matching caps is not found even when map returned here is not empty.